File: SnakeAndLadderGame/Player.cpp
//
//  Player.cpp
//  SnakeAndLadderGame
//
//  Created by Srijani Sen on 25/01/21.
//  Copyright © 2021 Srijani Sen. All rights reserved.
//

#include "Player.hpp"

Player::Player(std::string name, std::shared_ptr<Board>& boardPtr)
:
playerName(name),
currPosition(0),
mpBoardPtr(boardPtr)
{
}

bool Player::isPlayerTurn()
{
	return isTurn;
}

std::string Player::getName()
{
	return playerName;
}

int Player::setMove(int x)
{
	if(x + currPosition <= mpBoardPtr->getBoardRange())
	{
		currPosition += x;
		currPosition = mpBoardPtr->getNewPos(currPosition);
	}
	
	return currPosition;
}

int Player::getCurrPosition()
{
	return currPosition;
}

File: SnakeAndLadder/src/Player.java
public class Player {
    private String name;
    private int prevPosition;
    private int currPosition;

    public Player(String name){
        this.name = name;
        this.prevPosition = 0;
        this.currPosition = 0;
    }
    public String getName() {
        return name;
    }

    public int getPrevPosition() {
        return prevPosition;
    }

    public int getCurrPosition() {
        return currPosition;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setPrevPosition(int prevPosition) {
        this.prevPosition = prevPosition;
    }

    public void setCurrPosition(int currPosition) {
        this.currPosition = currPosition;
    }
}
